server/eclipse-project/src/main/java/us/freeandfair/corla/Main.java

```diff
@@ -80,24 +87,65 @@ public Main(final /*@ non_null */ Properties the_properties) {
   // Instance Methods
   
   /**
-   * Starts a ColoradoRLA server.
+   * Parse a port number from properties.
+   * 
+   * @param the_property The name of the property.
+   * @param the_default The default port number.
    */
-  public void start() {
-    LOGGER.info("starting server with properties: " + my_properties);
+  private int parsePortNumber(final String the_property, final int the_default) {
+    int result = the_default;
     
-    // get the port number from properties
-    int port_number = DEFAULT_PORT;
     try {
-      final int prop_port_number =
-          Integer.parseInt(my_properties.getProperty("port", String.valueOf(port_number)));
-      if (MIN_PORT <= prop_port_number && prop_port_number < MAX_PORT) {
-        port_number = prop_port_number;
+      final int prop_port =
+          Integer.parseInt(my_properties.getProperty(the_property, 
+                                                     String.valueOf(the_default)));
+      if (MIN_PORT <= prop_port && prop_port < MAX_PORT) {
+        result = prop_port;
       } else {
-        LOGGER.info("invalid port number in properties, using default port");
+        LOGGER.info("invalid port number in property " + the_property + 
+                    ", using default " + the_default);
       }
     } catch (final NumberFormatException e) {
-      LOGGER.info("could not read port number from properties, using default port");
+      LOGGER.info("could not read port number property " + the_property + 
+                  ", using default " + the_default);
+    }
+    
+    return result;
+  }
+  
+  /**
+   * Redirect a request from HTTP to HTTPS.
+   * 
+   * @param the_request The request.
+   * @param the_response The response.
+   * @param the_port The HTTPS port.
+   */
+  private void httpsRedirect(final Request the_request, final Response the_response,
```

Comment:
I thought about doing the latter, and even started writing it... but the URL class is immutable, and none of the URL constructors takes sufficient parameters for that to work universally (i.e., if the URL has a query or an anchor component - admittedly unlikely cases for this system - you can't pass those to a new URL constructor and they would be lost).